### ISSUE TITLE
Getting a "syntactically invalid" exception after import_module

### ISSUE THREAD

**Issue by ArneBachmannDLR** [beartype/beartype#171]
```
09.21./122016.453  ERROR    deployment:<listcomp>:120 |@beartyped generator logfiler.files.tailFile() return PEP 563-postponed type hint 'Generator[Optional[str], StreamCommand | Tuple[StreamCommand, object], None]' syntactically invalid (i.e., "unsupported operand type(s) for |: 'EnumMeta' and 'types.GenericAlias'") under:
09.21./122016.454  ERROR    deployment:<listcomp>:120 | ~~~~[ GLOBAL SCOPE ]~~~~
```

I think running on Python 3.9 doesn't accept the `|` type union, even if I use `from __future__ import annotations`.
Going back to `Union[...]` seems to resolve the issue, but this is not satisfactory.

**Comment by leycec** [comment]
Thanks so much for the fascinating issue, @ArneBachmannDLR. This is great stuff.

Sadly, your type hint in question *does* appear to be non-standard and thus at least semantically invalid – if not exactly syntactically invalid as @beartype suggests. [PEP 604-compliant `|`-style unions](https://peps.python.org/pep-0604) are only supported between combinations of simple classes and the `None` singleton (e.g., `int | str | None`). In this case, your type hint is erroneously attempting to create a `|`-style union `StreamCommand | Tuple[StreamCommand` between what *might* be a simple class (`StreamCommand`) and what definitely is *not* a simple class (`Tuple[StreamCommand, object]`).

In other words, PEP 604-compliant `|`-style unions do *not* obsolete standard PEP 484-compliant `typing.Union[...]` type hints; the former is just syntactic sugar for the latter in the edge case of simple types and the `None` singleton. But `typing.Union` is still the gold standard for general-purpose unions. It's hard to get more satisfactory than that, right?

Let us know if @beartype can do anything further here to improve your user experience. :+1:

**Comment by ArneBachmannDLR** [comment]
I simply assumed whatever mypy accepted is supposed to be valid...
But now that I switched back from `|` to `beartype.typing.Union` it works again.
Thanks for the detailed explanation, I simply assumed `|` is a full drop-in replacement for the pre-3.10 Union.

**Comment by leycec** [comment]
**No worries whatsoever.** Mypy can, indeed, behave untrustworthy. In my limited experience, a good >95% of mypy "errors" are just false positives that waste everyone's time. I'm so glad that we helped get your code sorted out, though. Typing standards like this can just be a :bear: to wrap your head around.

Personally, I'm fully agreement with your assumption: `|` really *should* have been a full drop-in replacement for the pre-3.10 `typing.Union`. The reason it isn't is that the authors of PEP 604 really wanted the `isinstance()` builtin to transparently support `|`-style type unions: e.g.,

```python
# Yup. This actually works. But does anyone care?
>>> isinstance('Pretty sure this is a string.', int | str | None)
True
>>> isinstance(b"Pretty sure this is ain't a string.", int | str | None)
False
```

Of course, once you do that, you've just implicitly constrained `|`-style type unions to simple types and `None`. If `|` supported anything more, then `isinstance()` itself would need to begin supporting arbitrarily complex PEP-complex type hints. CPython devs have *no* intention of going there, sadly.

Enter @beartype for the win. :wink:
